### PR TITLE
SNOW-1345421: Include pandas requirements in modin extra requirements.

### PR DESCRIPTION
**Pull request by sfc-gh-mvashishtha**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1345421

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   
    Include pandas requirements in modin extra requirements.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -29,6 +29,12 @@
 if os.getenv("SNOWFLAKE_IS_PYTHON_RUNTIME_TEST", False):
     REQUIRED_PYTHON_VERSION = ">=3.8"
 
+PANDAS_REQUIREMENTS = (
+    [
+        f"snowflake-connector-python[pandas]{CONNECTOR_DEPENDENCY_VERSION}",
```
Comment (sfc-gh-jkew) [inline setup.py:34]: It is weird to me that we require pandas through a transient dependency on the connector. Is that right?

Comment (sfc-gh-mvashishtha) [reply setup.py:34]: snowflake-snowpark-python[modin] gets pandas itself via the modin dependency as well. `snowflake-snowpark-python[pandas]` installs snowflake-connector-python together with the dependencies that the connector [needs](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-pandas#installation) to interoperate with pandas. Those [seem to be] both (https://github.com/snowflakedb/snowflake-connector-python/blob/eca86b9fcd5da9da722d18edc756c90ae940304b/setup.cfg#L95-L97) `pandas>=1.0.0,<3.0.0` and `pyarrow>=13.0.0`.

**Review by sfc-gh-helmeleegy — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -29,6 +29,12 @@
 if os.getenv("SNOWFLAKE_IS_PYTHON_RUNTIME_TEST", False):
     REQUIRED_PYTHON_VERSION = ">=3.8"
 
+PANDAS_REQUIREMENTS = (
```
Comment (sfc-gh-jdu) [inline setup.py:32]: Is it parenthesis for formatting, or actually it's a tuple? I guess PANDAS_REQUIREMENTS should just be a list here?

File: .github/workflows/precommit.yml [hunk 3]
```diff
@@ -411,6 +411,63 @@ jobs:
             .tox/.coverage
             .tox/coverage.xml
 
+  test-modin-extra-without-pandas-extra:
```
Comment (sfc-gh-jdu) [inline .github/workflows/precommit.yml:414]: Can this be in daily precommit tests?

**Review by sfc-gh-nkrishna — APPROVED** [review]

**Review by sfc-gh-azhan — COMMENTED** [review]
@sfc-gh-mvashishtha can you do a rebase once https://github.com/snowflakedb/snowpark-python/pull/1425 is in?

File: .github/workflows/precommit.yml (hunk 3 above)
Comment (sfc-gh-azhan) [reply .github/workflows/precommit.yml:414]: Can you move to the daily modin precommit once https://github.com/snowflakedb/snowpark-python/pull/1421/files is in?

**Review by sfc-gh-azhan — CHANGES_REQUESTED** [review]

File: setup.py (hunk 2 above)
Comment (sfc-gh-mvashishtha) [reply setup.py:32]: It's just for formatting. There's no comma after the list, so it's just a list, not a tuple of a list. the pre-commit hook likes to format it like this.

File: .github/workflows/precommit.yml (hunk 3 above)
Comment (sfc-gh-mvashishtha) [reply .github/workflows/precommit.yml:414]: Why should we move this to daily? It's just running a single test case, so it's fast.

Comment (sfc-gh-nkrishna) [reply .github/workflows/precommit.yml:414]: +1, I think we can keep this in

File: tests/integ/modin/test_modin_extra_without_pandas_extra.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+#
+
+import modin.pandas as pd
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from tests.integ.modin.sql_counter import sql_count_checker
+
+
+@sql_count_checker(query_count=1)
+def test_snow_1345421():
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/test_modin_extra_without_pandas_extra.py:12]:
```suggestion
def test_simple_to_pandas():
```

File: tox.ini [hunk 5]
```diff
@@ -101,6 +101,15 @@ deps =
     .[opentelemetry]
 commands = {env:SNOWFLAKE_PYTEST_DAILY_CMD} -vvv -m "integ or unit" {posargs:}  tests
 
+[testenv:modin_extra_without_pandas_extra]
+allowlist_externals = bash
+description = SNOW-1345421: test that we can use modin by specifying the modin extra but not the pandas extra.
+deps =
+    .[development]
```
Comment (sfc-gh-azhan) [inline tox.ini:108]: @sfc-gh-nkrishna we should make it more compact in your PR right?

File: .github/workflows/precommit.yml (hunk 3 above)
Comment (sfc-gh-azhan) [reply .github/workflows/precommit.yml:414]: sgtm if it is super fast.

**Review by sfc-gh-azhan — COMMENTED** [review]
Can we make sure those files are owned by both teams?

File: tox.ini (hunk 5 above)
Comment (sfc-gh-nkrishna) [reply tox.ini:108]: Yes

Comment (sfc-gh-mvashishtha) [reply tox.ini:108]: yeah after that PR the only `deps` here should be `[modin-development]`

**Comment by sfc-gh-mvashishtha** [comment]
>Can we make sure those files are owned by both teams?

@sfc-gh-nkrishna is doing that in #1425

File: setup.py (hunk 2 above)
Comment (sfc-gh-jdu) [reply setup.py:32]: hmm if I read correctly, isn't a comma after this list here (line 35)?

Comment (sfc-gh-mvashishtha) [reply setup.py]: Fixed.

**Review by sfc-gh-jdu — APPROVED** [review]

**Review by sfc-gh-azhan — APPROVED** [review]

**Pull request merged**